src/dialogs/more-info/controls/more-info-water_heater.js

```diff
@@ -0,0 +1,265 @@
+import '@polymer/iron-flex-layout/iron-flex-layout-classes.js';
+import '@polymer/paper-dropdown-menu/paper-dropdown-menu.js';
+import '@polymer/paper-item/paper-item.js';
+import '@polymer/paper-listbox/paper-listbox.js';
+import '@polymer/paper-toggle-button/paper-toggle-button.js';
+import { timeOut } from '@polymer/polymer/lib/utils/async.js';
+import { Debouncer } from '@polymer/polymer/lib/utils/debounce.js';
+import { html } from '@polymer/polymer/lib/utils/html-tag.js';
+import { PolymerElement } from '@polymer/polymer/polymer-element.js';
+
+import '../../../components/ha-water_heater-control.js';
+import '../../../components/ha-paper-slider.js';
+
+
+import attributeClassNames from '../../../common/entity/attribute_class_names.js';
+import featureClassNames from '../../../common/entity/feature_class_names';
+
+import EventsMixin from '../../../mixins/events-mixin.js';
+import LocalizeMixin from '../../../mixins/localize-mixin.js';
+
+/*
+ * @appliesMixin EventsMixin
+ * @appliesMixin LocalizeMixin
+ */
+class MoreInfoWaterHeater extends LocalizeMixin(EventsMixin(PolymerElement)) {
+  static get template() {
+    return html`
+    <style include="iron-flex"></style>
+    <style>
+      :host {
+        color: var(--primary-text-color);
+      }
+
+      .container-on,
+      .container-away_mode,
+      .container-temperature,
+      .container-operation_list,
+
+      .has-on .container-on,
+      .has-away_mode .container-away_mode,
+      .has-target_temperature .container-temperature,
+      .has-operation_mode .container-operation_list,
+
+      .container-operation_list iron-icon,
+
+      paper-dropdown-menu {
+        width: 100%;
+      }
+
+      paper-item {
+        cursor: pointer;
+      }
+
+      ha-paper-slider {
+        width: 100%;
+      }
+
+      ha-water_heater-control.range-control-left,
+      ha-water_heater-control.range-control-right {
+        float: left;
+        width: 46%;
+      }
+      ha-water_heater-control.range-control-left {
+        margin-right: 4%;
+      }
+      ha-water_heater-control.range-control-right {
+        margin-left: 4%;
+      }
+
+      .single-row {
+        padding: 8px 0;
+      }
+      }
+    </style>
+
+    <div class$="[[computeClassNames(stateObj)]]">
+
+      <template is="dom-if" if="[[supportsOn(stateObj)]]">
```

Comment:
You call `supportsOn` here but you don't have that method defined.

Reply:
Thanks, fixed.